zuul-core/src/main/java/com/netflix/zuul/message/http/HttpRequestMessageImpl.java

```diff
@@ -510,12 +514,23 @@ protected String generateInfoForLogging() {
     @Override
     public String getOriginalHost() {
         try {
-            return getOriginalHost(getHeaders(), getServerName());
+            if (!CACHE_ORIGINAL_HOST.get()) {
+                return generateOriginalHost(getHeaders(), getServerName());
```

Comment:
I'm not sure we need this FP to not-cache the original host; since it's intended for fetching the host on the inbound request, it shouldn't be mutated by any header processing, so keeping it always cached makes it simpler